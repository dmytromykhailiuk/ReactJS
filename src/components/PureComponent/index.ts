export { default as PureComponent } from "./PureComponent";
