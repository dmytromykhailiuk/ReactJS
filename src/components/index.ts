export * from "./Component";
export * from "./PureComponent";
export * from "./FunctionalComponent";
