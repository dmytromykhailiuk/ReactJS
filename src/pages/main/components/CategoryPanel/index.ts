export { default as CategoryPanel } from './CategoryPanel';
