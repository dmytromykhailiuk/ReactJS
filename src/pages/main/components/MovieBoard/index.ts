export { default as MovieBoard } from "./MovieBoard";
export { default as MovieBoardContainer } from "./MovieBoardContainer";
