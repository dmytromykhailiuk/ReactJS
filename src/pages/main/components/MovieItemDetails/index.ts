export { default as MovieItemDetails } from "./MovieItemDetails";
