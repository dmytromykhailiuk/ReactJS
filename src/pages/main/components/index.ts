export * from "./Banner";
export * from "./MovieBoard";
export * from "./SearchPanel";
export * from "./CategoryPanel";
export * from "./SortPanel";
export * from "./MovieList";
export * from "./MovieItem";
export * from "./MovieMenu";
export * from "./MovieModal";
