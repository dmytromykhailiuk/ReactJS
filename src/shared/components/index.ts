export * from "./Footer";
export * from "./Logo";
export * from "./Button";
export * from "./Input";
export * from "./ThreeDotsIcon";
export * from "./ErrorBoundary";
