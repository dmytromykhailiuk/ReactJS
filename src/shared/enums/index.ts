export * from "./button-types";
export * from "./button-size";
export * from "./categories";
export * from "./sorting-options";
export * from "./movie-form-values";
export * from "./router-paths";
export * from "./modal-types";
