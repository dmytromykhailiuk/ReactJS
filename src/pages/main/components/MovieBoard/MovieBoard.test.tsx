import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import { Button, Loader } from 'shared/components';
import { Categories, SortingOptionsProperties } from '../../../../shared/enums';
import MovieBoardView from './MovieBoardView';

describe('MovieBoardView', () => {
  it('should match snepshot', () => {
    const wrapper = shallow(
      <MovieBoardView
        movies={[{ id: 2 }] as any[]}
        movieBoardRef={{} as any}
        category={Categories.ALL}
        sortingOption={SortingOptionsProperties.RELEASE_DATE}
        isDownDirection
        moviesLoading={false}
        moviesAmount={10}
        moreMoviesLoaded
        loaded
        onEditMovie={() => {}}
        onDeleteMovie={() => {}}
        setCategory={() => {}}
        setSortingOption={() => {}}
        setIsDownDirectionValue={() => {}}
        showMoreMovies={() => {}}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should show 'No Movies Dound' message", () => {
    const wrapper = shallow(
      <MovieBoardView
        movies={[] as any[]}
        movieBoardRef={{} as any}
        category={Categories.ALL}
        sortingOption={SortingOptionsProperties.RELEASE_DATE}
        isDownDirection
        moviesLoading={false}
        moviesAmount={0}
        moreMoviesLoaded
        loaded
        onEditMovie={() => {}}
        onDeleteMovie={() => {}}
        setCategory={() => {}}
        setSortingOption={() => {}}
        setIsDownDirectionValue={() => {}}
        showMoreMovies={() => {}}
      />,
    );

    expect(wrapper.find('.movie-board__no-movie-found').text()).toEqual('No Movie Found');
  });

  it('should show Loader', () => {
    const wrapper = shallow(
      <MovieBoardView
        movies={[] as any[]}
        movieBoardRef={{} as any}
        category={Categories.ALL}
        sortingOption={SortingOptionsProperties.RELEASE_DATE}
        isDownDirection
        moviesLoading
        moviesAmount={10}
        moreMoviesLoaded
        loaded={false}
        onEditMovie={() => {}}
        onDeleteMovie={() => {}}
        setCategory={() => {}}
        setSortingOption={() => {}}
        setIsDownDirectionValue={() => {}}
        showMoreMovies={() => {}}
      />,
    );
    expect(wrapper.exists(Loader)).toBeTruthy();
  });

  it("should not show 'more movies' button", () => {
    const wrapper = shallow(
      <MovieBoardView
        movies={[{ id: 2 }] as any[]}
        movieBoardRef={{} as any}
        category={Categories.ALL}
        sortingOption={SortingOptionsProperties.RELEASE_DATE}
        isDownDirection
        moviesLoading={false}
        moviesAmount={10}
        moreMoviesLoaded={false}
        loaded
        onEditMovie={() => {}}
        onDeleteMovie={() => {}}
        setCategory={() => {}}
        setSortingOption={() => {}}
        setIsDownDirectionValue={() => {}}
        showMoreMovies={() => {}}
      />,
    );
    expect(wrapper.exists(Button)).toBeFalsy();
  });
});
