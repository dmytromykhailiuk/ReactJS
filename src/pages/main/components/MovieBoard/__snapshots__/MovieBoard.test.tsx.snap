// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieBoardView should match snepshot 1`] = `
<div
  className="movie-board"
>
  <div
    className="movie-board__header"
  >
    <CategoryPanel
      onChangeCategory={[Function]}
      selectedCategory="ALL"
    />
    <SortPanel
      isDownDirection={true}
      onChangeSortingDirection={[Function]}
      setSortingOption={[Function]}
      sortingOption="release_date"
    />
    <div
      className="movie-board__header-underline"
    />
  </div>
  <h2
    className="movie-board__count"
  >
    <span
      className="movie-board__count-number"
    >
      10
    </span>
    Â  movie
    s
     found
  </h2>
  <MovieList
    movies={
      Array [
        Object {
          "id": 2,
        },
      ]
    }
    onDeleteMovie={[Function]}
    onEditMovie={[Function]}
  />
  <div
    className="movie-board__show-more-button"
  >
    <Button
      onButtonClicked={[Function]}
      type="secondary"
    >
      SHOW MORE
    </Button>
  </div>
</div>
`;
